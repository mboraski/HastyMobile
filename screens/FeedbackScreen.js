// Third Party Imports
import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { connect } from 'react-redux';

// Relative Imports
import CloseButton from '../components/CloseButton';
import DoneButton from '../components/DoneButton';
import Rating from '../components/Rating';
import Color from '../constants/Color';
import Style from '../constants/Style';
import { emY } from '../utils/em';

class FeedbackScreen extends Component {
    state = {
        userRating: 0,
        productRating: 0,
        overallRating: 0
    };

    handleUserRating = userRating => {
        this.setState({ userRating });
    };

    handleProductRating = productRating => {
        this.setState({ productRating });
    };

    handleOverallRating = overallRating => {
        this.setState({ overallRating });
    };

    render() {
        const { name, numProducts } = this.props;
        const { userRating, productRating, overallRating } = this.state;
        const productTitle = numProducts > 1 ? 'How were your products?' : 'How was your product?';
        return (
            <View style={styles.container}>
                <Text style={styles.title}>How was {name}?</Text>
                <Rating style={styles.rating} value={userRating} onChange={this.handleUserRating} />
                <Text style={styles.title}>{productTitle}</Text>
                <Rating
                    style={styles.rating}
                    value={productRating}
                    onChange={this.handleProductRating}
                />
                <Text style={styles.title}>How was your experience overall?</Text>
                <Rating
                    style={styles.rating}
                    value={overallRating}
                    onChange={this.handleOverallRating}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        backgroundColor: Color.WHITE,
        paddingTop: emY(2)
    },
    title: {
        marginBottom: emY(1),
        fontSize: emY(1)
    },
    rating: {
        marginBottom: emY(4)
    }
});

FeedbackScreen.navigationOptions = {
    title: 'Feedback',
    headerLeft: <CloseButton />,
    headerRight: <DoneButton />,
    headerStyle: Style.header,
    headerTitleStyle: Style.headerTitle
};

const mapStateToProps = state => ({
    name: 'Jessica',
    numProducts: 1
});

export default connect(mapStateToProps, {})(FeedbackScreen);
