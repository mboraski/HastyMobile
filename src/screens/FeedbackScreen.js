// Third Party Imports
import React, { Component } from 'react';
import { StyleSheet, View, Text, KeyboardAvoidingView } from 'react-native';
import { connect } from 'react-redux';
import { Button } from 'react-native-elements';

// Relative Imports
import { showFeedbackForm, hideFeedbackForm } from '../actions/uiActions';
import FeedbackFormCloseButton from '../containers/FeedbackFormCloseButton';
import Rating from '../components/Rating';
import FeedbackForm from '../containers/FeedbackForm';
import Color from '../constants/Color';
import Style from '../constants/Style';
import { emY } from '../utils/em';

const keyboardVerticalOffset = emY(1);

class FeedbackScreen extends Component {
    state = {
        userRating: 0,
        productRating: 0,
        overallRating: 0
    };

    componentWillUnmount() {
        this.props.hideFeedbackForm();
    }

    handleUserRating = userRating => {
        this.setState({ userRating });
    };

    handleProductRating = productRating => {
        this.setState({ productRating });
    };

    handleOverallRating = overallRating => {
        this.setState({ overallRating });
    };

    onButtonPress = () => {
        if (!this.props.feedbackFormVisible) {
            if (
                this.state.userRating <= 3 ||
                this.state.productRating <= 3 ||
                this.state.overallRating <= 3
            ) {
                this.props.showFeedbackForm();
            }
        }
    };

    onSubmitSuccess = () => {
        this.props.navigation.goBack();
    };

    render() {
        const { name, numProducts, feedbackFormVisible } = this.props;
        const { userRating, productRating, overallRating } = this.state;
        const productTitle = numProducts > 1 ? 'How were your products?' : 'How was your product?';
        return feedbackFormVisible ? (
            <KeyboardAvoidingView
                style={styles.container}
                behavior="padding"
                keyboardVerticalOffset={keyboardVerticalOffset}
            >
                <FeedbackForm onSubmitSucceeded={this.onSubmitSuccess} />
            </KeyboardAvoidingView>
        ) : (
            <View style={styles.container}>
                <View style={styles.ratings}>
                    <Text style={styles.title}>How was {name}?</Text>
                    <Rating
                        style={styles.rating}
                        value={userRating}
                        onChange={this.handleUserRating}
                    />
                    <Text style={styles.title}>{productTitle}</Text>
                    <Rating
                        style={styles.rating}
                        value={productRating}
                        onChange={this.handleProductRating}
                    />
                    <Text style={styles.title}>How was your experience overall?</Text>
                    <Rating
                        style={styles.rating}
                        value={overallRating}
                        onChange={this.handleOverallRating}
                    />
                </View>
                <Button
                    title="SEND"
                    onPress={this.onButtonPress}
                    containerViewStyle={styles.buttonContainer}
                    buttonStyle={styles.button}
                    textStyle={styles.buttonText}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Color.WHITE
    },
    title: {
        marginBottom: emY(1),
        fontSize: emY(1)
    },
    ratings: {
        alignItems: 'center',
        flex: 1,
        paddingTop: emY(2)
    },
    rating: {
        marginBottom: emY(4)
    },
    button: {
        backgroundColor: '#000',
        paddingVertical: emY(1),
        marginBottom: emY(1)
    },
    buttonText: {
        fontSize: emY(1)
    }
});

FeedbackScreen.navigationOptions = ({ navigation }) => {
    const handlePressClose = () => navigation.goBack();
    return {
        title: 'Feedback',
        headerLeft: <FeedbackFormCloseButton onPress={handlePressClose} />,
        headerStyle: Style.header,
        headerTitleStyle: Style.headerTitle
    };
};

const mapStateToProps = state => ({
    name: 'Jessica',
    numProducts: 1,
    feedbackFormVisible: state.ui.feedbackFormVisible
});

const mapDispatchToProps = {
    showFeedbackForm,
    hideFeedbackForm
};

export default connect(mapStateToProps, mapDispatchToProps)(FeedbackScreen);
