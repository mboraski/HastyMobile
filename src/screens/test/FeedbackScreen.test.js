import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import { FeedbackScreen } from '../FeedbackScreen';

describe('FeedbackScreen', () => {
    it('renders correctly', () => {
        const props = {
            name: 'Jessica',
            numProducts: 1,
            feedbackFormVisible: true
        };
        const render = shallow(<FeedbackScreen {...props} />);
        expect(render).toMatchSnapshot();
    });
});
