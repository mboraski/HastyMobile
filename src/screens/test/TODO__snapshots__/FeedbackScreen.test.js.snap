// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackScreen renders correctly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={16}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <ReduxForm
    onSubmitSucceeded={[Function]}
  />
</KeyboardAvoidingView>
`;
